subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/attributes/ArtifactViewAttributesIntegrationTest.groovy

```diff
@@ -0,0 +1,382 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.integtests.resolve.attributes
+
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+import spock.lang.Ignore
+
+/**
+ * Tests for [org.gradle.api.artifacts.ArtifactView ArtifactView] that ensure it uses the "live" attributes
+ * of the configuration that created it to select files and artifacts.
+ *
+ * These tests assume a situation where a consumer should resolve the secondary "classes" variant of a producer
+ * and obtain the "main" directory as the only file.  As the necessary attributes are not present on the
+ * consumer's compileClasspath configuration until the beforeLocking callback method is called, the previous
+ * incorrect behavior of the ArtifactView was to select the standard jar file variant instead of the "classes"
+ * variant.
+ */
+class ArtifactViewAttributesIntegrationTest extends AbstractIntegrationSpec {
+    private static declareIncomingFiles = "def incomingFiles = configurations.compileClasspath.incoming.files"
+    private static declareIncomingArtifacts = "def incomingArtifacts = configurations.compileClasspath.incoming.artifacts"
+    private static declareArtifactViewFiles = "def artifactViewFiles = configurations.compileClasspath.incoming.artifactView { }.files"
+    private static declareArtifactViewArtifacts = "def artifactViewArtifacts = configurations.compileClasspath.incoming.artifactView { }.artifacts"
+
+    private static iterateIncomingFiles = """logger.warn 'Incoming Files:'
```

Comment:
Is this iteration code necessary for the tests to function? Otherwise IMO they just clutter the tests

Reply:
Yes, the iteration causes the resolution to happen.

Reply:
We should be able to trigger resolution without iteration and printing. The `filesComparison` and `artifactComparison` are in reality iterating with the `*.` syntax

Reply:
We need to trigger the resolution prior to the comparison to demonstrate the change in behavior, otherwise the tests pass without the change to `DefaultConfiguration`.  

The print output is just helpful to see if a test fails.  I could trigger resolve another way, but I found the output helpful when working through the problem and verifying the solution.  I don't see a big benefit to removing it - it's already absent from the bodies of the test methods, which just use the template variables.